apis/tempo/v1alpha1/microservices_types.go

```diff
@@ -86,6 +93,188 @@ type MicroservicesStatus struct {
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Authorization"
+	Authorization *AuthorizationSpec `json:"authorization,omitempty"`
+}
+
+// PermissionType is a Tempo Gateway RBAC permission.
+//
+// +kubebuilder:validation:Enum=read;write
+type PermissionType string
+
+const (
+	// Write gives access to write data to a tenant.
+	Write PermissionType = "write"
+	// Read gives access to read data from a tenant.
+	Read PermissionType = "read"
+)
+
+// SubjectKind is a kind of Tempo Gateway RBAC subject.
+//
+// +kubebuilder:validation:Enum=user;group
+type SubjectKind string
+
+const (
+	// User represents a subject that is a user.
+	User SubjectKind = "user"
+	// Group represents a subject that is a group.
+	Group SubjectKind = "group"
+)
+
+// Subject represents a subject that has been bound to a role.
+type Subject struct {
+	Name string      `json:"name"`
+	Kind SubjectKind `json:"kind"`
+}
+
+// RoleBindingsSpec binds a set of roles to a set of subjects.
+type RoleBindingsSpec struct {
+	Name     string    `json:"name"`
+	Subjects []Subject `json:"subjects"`
+	Roles    []string  `json:"roles"`
+}
+
+// AuthorizationSpec defines the opa, role bindings and roles
+// configuration per tenant for tempo Gateway component.
+type AuthorizationSpec struct {
+	// OPA defines the spec for the third-party endpoint for tenant's authorization.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OPA Configuration"
+	OPA *OPASpec `json:"opa"`
+	// Roles defines a set of permissions to interact with a tenant.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Static Roles"
+	Roles []RoleSpec `json:"roles"`
+	// RoleBindings defines configuration to bind a set of roles to a set of subjects.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Static Role Bindings"
+	RoleBindings []RoleBindingsSpec `json:"roleBindings"`
+}
+
+// RoleSpec describes a set of permissions to interact with a tenant.
+type RoleSpec struct {
+	Name        string           `json:"name"`
+	Resources   []string         `json:"resources"`
+	Tenants     []string         `json:"tenants"`
+	Permissions []PermissionType `json:"permissions"`
+}
+
+// OPASpec defines the opa configuration spec for Tempo Gateway component.
+type OPASpec struct {
+	// URL defines the third-party endpoint for authorization.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="OpenPolicyAgent URL"
+	URL string `json:"url"`
+}
+
+// TenantSecretSpec is a secret reference containing name only
+// for a secret living in the same namespace as the (Tempo) Microservices custom resource.
+type TenantSecretSpec struct {
```

Comment:
can be removed for now

Reply:
What is it used for?

Reply:
`Secret defines the spec for the clientID, clientSecret and issuerCAPath for tenant's authentication` its there to folllow the structure of the loki-operator. But we could also replace it with a [v1.Secret](https://kubernetes.io/docs/concepts/configuration/secret/).